frontends/pytorch/csrc/c10_dispatch/acap_dispatch.cpp

```diff
@@ -127,6 +198,66 @@ void AcapController::fallbackKernel(const OperatorHandle &opHandle,
   current->fallbackKernelImpl(opHandle, stack);
 }
 
+at::Tensor AcapController::convolutionKernel(
+    const at::Tensor &input, const at::Tensor &weight,
+    const c10::optional<at::Tensor> &bias, const at::IntArrayRef stride,
+    const at::IntArrayRef padding, const at::IntArrayRef dilation,
+    const bool transposed, const at::IntArrayRef output_padding,
+    const int64_t groups) {
+  static c10::OperatorName opName{"aten::convolution", ""};
+  auto &dispatcher = c10::Dispatcher::singleton();
+  auto opHandle = dispatcher.findOp(opName);
+  assert(opHandle && "could not find convolution op");
+  auto opTyped = opHandle->typed<at::Tensor(
+      const at::Tensor &, const at::Tensor &, const c10::optional<at::Tensor> &,
+      const at::IntArrayRef, const at::IntArrayRef, const at::IntArrayRef,
+      const bool, const at::IntArrayRef, const int64_t)>();
+
+  // Exclude recursive calls: convolution is completely emitted by this
+  // kernel.
+  c10::DispatchKeySet keySet{kAcapDispatchKey, kAcapGradDispatchKey};
+  c10::impl::ExcludeDispatchKeyGuard exclusion(keySet);
+
+  auto current = getCurrent();
+  if (!current) {
```

Comment:
is there a deeper explanation of the `if (!current)` besides "we're just not capturing right now?".

Reply:
That is the explanation. Left a comment on getCurrentThreadAcapController().